plugins/modules/ntnx_acps.py

```diff
@@ -0,0 +1,449 @@
+        user_uuids=dict(type="list", elements="str"),
+        user_group_uuids=dict(type="list", elements="str"),
+        role=dict(
+            type="dict", options=entity_by_spec, mutually_exclusive=mutually_exclusive
+        ),
+        filters=dict(type="list", elements="dict", options=filter_spec),
+    )
+
+    return module_args
+
+
+def create_acp(module, result):
+    acp = ACP(module)
+    spec, error = acp.get_spec()
+    if error:
+        result["error"] = error
+        module.fail_json(msg="Failed generating acp spec", **result)
+
+    if module.check_mode:
+        result["response"] = spec
+        return
+
+    resp = acp.create(spec)
+    acp_uuid = resp["metadata"]["uuid"]
+    result["changed"] = True
+    result["response"] = resp
+    result["acp_uuid"] = acp_uuid
+    result["task_uuid"] = resp["status"]["execution_context"]["task_uuid"]
+
+    if module.params.get("wait"):
+        wait_for_task_completion(module, result)
+        resp = acp.read(acp_uuid)
+        result["response"] = resp
+
+
+def update_acp(module, result):
+    acp_uuid = module.params["acp_uuid"]
+    state = module.params.get("state")
+
+    acp = ACP(module)
+    resp = acp.read(acp_uuid)
+    result["response"] = resp
+    utils.strip_extra_attrs_from_status(resp["status"], resp["spec"])
+    resp.pop("status")
+
+    spec, error = acp.get_spec(resp)
+
+    if error:
+        result["error"] = error
+        module.fail_json(msg="Failed generating ACP spec", **result)
+
+    if module.check_mode:
+        result["response"] = spec
+        return
+
+    if utils.check_for_idempotency(spec, resp, state=state):
+        result["skipped"] = True
+        module.exit_json(msg="Nothing to change")
+
+    resp = acp.update(spec, acp_uuid)
+    acp_uuid = resp["metadata"]["uuid"]
+    result["changed"] = True
+    result["response"] = resp
+    result["acp_uuid"] = acp_uuid
+    result["task_uuid"] = resp["status"]["execution_context"]["task_uuid"]
+
+    if module.params.get("wait"):
+        wait_for_task_completion(module, result)
+        resp = acp.read(acp_uuid)
+        result["response"] = resp
+
+
+def delete_acp(module, result):
+    acp_uuid = module.params["acp_uuid"]
+    if not acp_uuid:
+        result["error"] = "Missing parameter acp_uuid in playbook"
+        module.fail_json(msg="Failed deleting acp", **result)
+
+    acp = ACP(module)
+    resp = acp.delete(acp_uuid)
+    result["changed"] = True
+    result["response"] = resp
+    result["acp_uuid"] = acp_uuid
+    result["task_uuid"] = resp["status"]["execution_context"]["task_uuid"]
+
+    if module.params.get("wait"):
+        wait_for_task_completion(module, result)
+
+
+def wait_for_task_completion(module, result):
+    task = Task(module)
+    task_uuid = result["task_uuid"]
+    resp = task.wait_for_completion(task_uuid)
+    result["response"] = resp
+
+
+def run_module():
+    module = BaseModule(argument_spec=get_module_spec(), supports_check_mode=True)
```

Comment:
Add mutually exclusive and required if for having validations in update, create and delete scenario. Look into module for images